CMakeLists.txt

```diff
@@ -11,6 +11,10 @@ set(CMAKE_MODULE_PATH
 
 include(cmake/utils.cmake)
 
+# Set CFLAGS/CXXFLAGS
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror -std=gnu11")
```

Comment:
Maybe enable -Werror only on debug build, just like in tarantool?

Reply:
Not sure it's a good idea. Tarantool has separate target for Debug build in CI but we don't have that for modules.

I don't see nothing bad to show and prevent as much warnings as possible.

Reply:
1. You can run `cmake . -DCMAKE_BUILD_TYPE=Debug|Release|RelWithDebInfo && make VERBOSE=1` and see different compilation flags.
2. https://github.com/tarantool/tarantool/blob/efc30ccf8393a1aad19d4c17299842899d003caa/cmake/compiler.cmake#L372-L373 (it comes from this [commit](https://github.com/tarantool/tarantool/commit/7fa4be57aa540b995909150f4026d689c661b80d) and this [issue](https://bugs.launchpad.net/tarantool/+bug/805350)).

Reply:
Yes, we can, but I think it's more relevant for Tarantool with default Debug mode. But I'm not sure it's so relevant for modules.

E.g. in avro-schema:  https://github.com/tarantool/avro-schema/blob/3b2d264e1f7828dda034aa5cae30184f2b9adbfa/CMakeLists.txt#L21-L23

Reply:
Proportionally to amount of code.

Default cmake flags for different build types are different.